src/components/editor/index.tsx

```diff
@@ -19,12 +18,26 @@ import {
 } from '@amoutonbrady/solid-heroicons/outline';
 import { liftOff } from './setupSolid';
 
+const createUpdater: () => [<T>(a: T) => T, () => void] = () => {
+  const [x, setX] = createSignal(0, () => false);
+  return [
+    (a) => {
+      x();
+      return a;
+    },
+    () => {
+      setX(x() + 1);
+    },
+  ];
+};
```

Comment:
Basically this creates a signal, so that we can trigger effects arbitrarily. Basically what is important is the usage: we can create an updater function that calls the createEffects that read from model(). 

A note: I realized there might be a way to get rid of this by just exposing the functions inside createEffect and calling those functions directly. If you are interested, I can try to work that into this